diff --git a/dwlmsg.c b/dwlmsg.c
index ea2ff04..879ea0a 100644
--- a/dwlmsg.c
+++ b/dwlmsg.c
@@ -112,6 +112,9 @@ static void
 dwl_ipc_output_layout(void *data, struct zdwl_ipc_output_v2 *dwl_ipc_output,
 	uint32_t layout)
 {
+	if (!(lflag && mode & GET)) return;
+	if (data) printf("%s ", (char *)data);
+	printf("layout_index %u\n", layout);
 }
 
 static void
@@ -253,6 +256,12 @@ dwl_ipc_output_frame(void *data, struct zdwl_ipc_output_v2 *dwl_ipc_output)
 	fflush(stdout);
 }
 
+static void
+dwl_ipc_output_focused_geometry(void *data, struct zdwl_ipc_output_v2 *dwl_ipc_output,
+	int32_t x, int32_t y, int32_t width, int32_t height)
+{
+}
+
 static const struct zdwl_ipc_output_v2_listener dwl_ipc_output_listener = {
 	.toggle_visibility = dwl_ipc_output_toggle_visibility,
 	.active = dwl_ipc_output_active,
@@ -264,6 +273,7 @@ static const struct zdwl_ipc_output_v2_listener dwl_ipc_output_listener = {
 	.fullscreen = dwl_ipc_output_fullscreen,
 	.floating = dwl_ipc_output_floating,
 	.frame = dwl_ipc_output_frame,
+	.focused_geometry = dwl_ipc_output_focused_geometry,
 };
 
 static void
diff --git a/protocols/dwl-ipc-unstable-v2.xml b/protocols/dwl-ipc-unstable-v2.xml
index 0a6e7e5..07cc1c2 100644
--- a/protocols/dwl-ipc-unstable-v2.xml
+++ b/protocols/dwl-ipc-unstable-v2.xml
@@ -177,5 +177,15 @@ I would probably just submit raphi's patchset but I don't think that would be po
       </description>
       <arg name="is_floating" type="uint" summary="If the selected client is floating. Nonzero is valid, zero invalid"/>
     </event>
+
+    <event name="focused_geometry" since="2">
+      <description summary="Send focused client geometry">
+          Sends the focused client's geometry (x, y, width, and height) to the client.
+      </description>
+      <arg name="x" type="int" summary="X coordinate of the focused client"/>
+      <arg name="y" type="int" summary="Y coordinate of the focused client"/>
+      <arg name="width" type="int" summary="Width of the focused client"/>
+      <arg name="height" type="int" summary="Height of the focused client"/>
+    </event>
   </interface>
 </protocol>
